routes/merged/project/location/vm.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get api? do
+      vm_list_api_response(vm_list_dataset)
+    end
+
+    r.on NAME_OR_UBID do |vm_name, vm_ubid|
+      if vm_name
+        r.post api? do
+          vm_post(vm_name)
+        end
+
+        filter = {Sequel[:vm][:name] => vm_name}
+      else
+        filter = {Sequel[:vm][:id] => UBID.to_uuid(vm_ubid)}
+      end
+
+      filter[:location] = @location
+      vm = @project.vms_dataset.first(filter)
+
+      unless vm
+        response.status = r.delete? ? 204 : 404
+        r.halt
+      end
+
+      r.get true do
+        Authorization.authorize(current_account.id, "Vm:view", vm.id)
+        @vm = Serializers::Vm.serialize(vm, {detailed: true})
+        if api?
+          @vm
+        else
+          view "vm/show"
+        end
+      end
+
+      r.delete true do
+        Authorization.authorize(current_account.id, "Vm:delete", vm.id)
+        vm.incr_destroy
+        response.status = 204
+        nil
+      end
+    end
+
+    # 204 response for invalid names
+    r.is String do |vm_name|
+      r.post do
+        vm_post(vm_name)
+      end
```

Comment:
General Ruby styling question: What's your opinion on using one-line blocks for single-line codes? Are they popular in Ruby?

```suggestion
      r.post { vm_post(vm_name) }
```

Reply:
I think using single line blocks is fine for routes.